src/averageAges.js

```diff
@@ -55,7 +82,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(person =>
+      person.sex === 'm' && people.some(woman =>
+        woman.name === person.mother))
+    : people.filter(person =>
+      people.some(woman => woman.name === person.mother));
```

Comment:
Why to duplicate filter method usage, use it once and select needed condition by ternary operator